### ISSUE TITLE
Handle HEIC/HEIF format in applicable places

### ISSUE THREAD

**Issue by eksperts** [fusetools/fuselibs-public#668]
Made aware of this on community: https://www.macworld.co.uk/feature/iphone/what-is-heic-3660408/

With iOS 11, Apple have started using a relatively new file format for images, which we currently don't support. Consequently, when trying to resize images with the `.heic` (`.heif`?) extension, Fuse fails with `ImageSource error: ‘BundleFileImageSource-failed-conversion`, which is not surprising at all.

We should either add support for that file format, or implement a workaround that converts the image to a supported format on-the-fly. The latter seems very wrong.

**Comment by kusma** [comment]
I don't think this has anything to do with resizing, but with displaying the result. The reason for this is the ImageSource-part of the error-message. ImageTools (which does the resizing) does not use `ImageSource`.

**Comment by ruudvanham** [comment]
Is there a temporary workarround to support this kind of images within our app? Is it possible to convert the .heic img to jpg before showing of some kind.

**Comment by ruudvanham** [comment]
@kusma Seems to work, thanks!

**Comment by chrfalch** [comment]
The simplest solution is to rename the resulting file with a JPEG extension instead of the provided HEIC extension. 

In `ImageHelper.m` in the iOS part of FUSE the method `saveImage` converts PNG images to PNG and everything else to JPEG. This is ok and saves us from the HEIC problem. But the method misses renaming the extension from HEIC to JPEG - it just ends up returning the same path which won't work in Fuse. Here is our code that solves the problem from within JavaScript:

```javascript

var FuseEnvironment = require('FuseJS/Environment');
var FileSystem = require("FuseJS/FileSystem");

if(imagePath.toLowerCase().endsWith(".heic") && FuseEnvironment.ios){
    // The image is now converted (in Camera Roll) to JPEG if it is in HEIC format.
    // We'll just make sure the path is correct by renaming it to jpg.
    var newPath = imagePath.replace(".heic", ".jpg");
    newPath = newPath.replace(".HEIC", ".jpg");
    FileSystem.moveSync(image.path, newPath);
  // newPath now points to the correct path.
```

**Comment by eksperts** [comment]
Indeed, the root cause of the issue is the code not renaming the resulting image.
A fix for the `ImageHelper.m` file can be found here: https://github.com/eksperts/fuselibs-public/commit/41820927ab12c21c41a6039f5216d907e4d957ec
